src/receive_imf.rs

```diff
@@ -288,6 +288,9 @@ pub(crate) async fn receive_imf_inner(
         received_msg = None;
     }
 
+    let verified_encryption =
+        has_verified_encryption(context, &mime_parser, from_id, &to_ids).await?;
```

Comment:
It's unrelated, but looks like has_verified_encryption() isn't a good function name if it doesn't return `bool`